### PR TITLE
more compatibility with pyxattr for bup

### PR DESCRIPTION
**Pull request by jmberg**
With this code, I can pass the bup test suite with xattr instead of pyxattr.

The list() functions is very ugly but actually used there, so there isn't any choice if it should be API compatible (and otherwise the whole point of this pull request is moot.)

### CODE REVIEW

File: xattr/__init__.py [hunk 1]
```diff
@@ -160,30 +160,49 @@ def itervalues(self):
...
-def setxattr(f, attr, value, options=0, symlink=False):
-    if symlink:
+def get_all(f, symlink=False, nofollow=False):
+    symlink = symlink or nofollow
+    options = symlink and XATTR_NOFOLLOW or 0
+    return xattr(f).items(options=options)
+
+# XXX: This is really ugly - now we cannot use the standard python
+#      list() anymore in this module!
+_pylist = list
```
Comment (etrepum) [inline xattr/__init__.py]: This is horrible, what a bad API choice :(

This should definitely be defined at the top, before any of the functions.

File: xattr/__init__.py [hunk 2]
```diff
@@ -160,30 +160,49 @@ def itervalues(self):
...
+def list(f, symlink=False, nofollow=False):
+    symlink = symlink or nofollow
+    return xattr(f).list(options=symlink and XATTR_NOFOLLOW or 0)
```
Comment (etrepum) [inline xattr/__init__.py]: How does this differ from listxattr?

File: xattr/__init__.py [hunk 3]
```diff
@@ -160,30 +160,49 @@ def itervalues(self):
...
+    symlink = symlink or nofollow
     return tuple(xattr(f).list(options=symlink and XATTR_NOFOLLOW or 0))
 
 
-def getxattr(f, attr, symlink=False):
+def getxattr(f, attr, symlink=False, nofollow=False):
+    symlink = symlink or nofollow
     return xattr(f).get(attr, options=symlink and XATTR_NOFOLLOW or 0)
 
+get = getxattr
```
Comment (etrepum) [inline xattr/__init__.py]: I think it would be better to put all of these compatibility aliases in one place at the bottom.

**Comment by etrepum** [comment]
Maybe it would be better to just submit a PR to bup to switch to xattr? xattr is 131 times more popular than pyxattr.

**Comment by jmberg** [comment]
Will put a new pull request.

**Pull request closed without merging**